numpy/core/arrayprint.py

```diff
@@ -434,10 +432,9 @@ def array2string(a, max_line_width=None, precision=None,
 
         The length of the prefix string is used to align the
         output correctly.
-    style : _NoValue, optional
-        Has no effect, do not use.
-
-        .. deprecated:: 1.14.0
+    style : function, optional
+        A function that accepts an ndarray and returns a string.  Used only
+        when the shape of `a` is equal to ``()``, i.e. for 0-D arrays.
```

Comment:
Correct.  This is actually the original docstring (from before any of my changes).

By the way, I structured this PR in two commits: A revert commit of my recent 0d changes, then a new commit for this PR. 

The diff of the second commit show the changes of this PR relative to numpy 1.13, where you can see th docstring is unchanged.